### DISCUSSION TITLE
Welcome to heizoel24mex Discussions!

### DISCUSSION THREAD

**Discussion by MartinStoffel** [MartinStoffel/heizoel24mex#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by LennyM8472** [comment]
Hi can you add additional sensors.
Like last paid pprce, current price and price forecast

**Reply by MartinStoffel** [reply]
The interface I reverse engineered does not provide those values. All values provided are visible as attribue on the Sensor MEX Sensor ID. 

You can easily creat them with configutartion see: https://www.home-assistant.io/integrations/template/

**Comment by itsmicash** [comment]
@MartinStoffel 
Hallo Martin,

I installed the MEX device 2 days ago. the MEX app said it needs around 6 days to calibrate. so i don‘t expect any volume changes in this period and keep on waiting.
when is your integration pulling the data from the internet? is there a schedule or does it receive a push?

Because i noticed that i had to do the calibration again as my IBC adapter was not tight enough, that changed the height of the sensor, so the new volume calculation was different. but your integration still reflected the volume from the first calibration.

LG Micha

**Reply by MartinStoffel** [reply]
Hi Micha

The data is read every 12 hours. The height measurement can be triggered by the button and needs no calibration, to my knowledge.

The calibration is probably needed for the rates, like liters/day or so.

**Reply by itsmicash** [reply]
Hi @MartinStoffel 

Ok but i understand that the sensor needs 6 days of tracking  the usage before it updates the sensor data daily. At least thats what was described when setting up the sensor inside the mobile app.
Because after 2 days it still displays the same number of liters left, even though im heating the house a lot.

I always compare what the MEX app displays in comparison to the integration to validate if everything is working fine.
I believe that after 6 days the app will display daily results, but i am not sure. For now i see the same volume in app and HA since calibration. 

is the integration pulling the data if the last_changed == 12 hours? so basically if i reload the integration in HA or push the button it will count 12 hours from that time on? Could it be that the mobile app displays the latest oil stock earlier than the integration?

**Reply by MartinStoffel** [reply]
Hello @itsmicash 

The sensor sends it's reading once a day, when it did so is also visible on the app (top right of the screen). The resolution of the sensor is basically 1cm (it measures the fill level, distance from sensor to oil surface), so you will only see a change after, whatever number of liters corresponds to the height of 1cm in your tank is consumed. And then your fill level will not change by 1 litwr but by several liters. Depending on the shape of your tank, this is not even linear, meaning if it's round, then 1cm change means a different volume in liters, depending of the current fill level.

What is another interesting fact, since it measures the fill level, and has 1cm resolutuion, it can happen, that the reading goes also up due to the fact, that oil expands/shrinks with temperature change. 

For the timing - yes, it could well be, that the app displays the data before you see it on mobile alerts. A restart of home assistant will get the latest reading.

**Comment by itsmicash** [comment]
@MartinStoffel 
thanks for the clear explanation, i saw in the app today that the fill level was reduced by 1 cm.
Maybe one more question:
According to the integration, the last pull was 19 hours ago, which is more than expected 12 hours.
do i need to set up an automation to restart HA, to refresh the values?
![IMG_1898](https://github.com/user-attachments/assets/cdca41c7-c11a-4517-939c-f244a2d16360)

**Reply by MartinStoffel** [reply]
When there is no difference from the previous reading, then there is no update shown.
